Code/Framework/AzFramework/AzFramework/Physics/CharacterBus.h

```diff
@@ -103,4 +104,75 @@ namespace Physics
-} // namespace Physics
+
+    /// Messages sent by character controllers.
+    class CharacterNotifications : public AZ::ComponentBus
+    {
+    private:
+        template<class Bus>
+        struct CharacterNotificationsConnectionPolicy : public AZ::EBusConnectionPolicy<Bus>
+        {
+            static void Connect(
+                typename Bus::BusPtr& busPtr,
+                typename Bus::Context& context,
+                typename Bus::HandlerNode& handler,
+                typename Bus::Context::ConnectLockGuard& connectLock,
+                const typename Bus::BusIdType& id = 0)
+            {
+                AZ::EBusConnectionPolicy<Bus>::Connect(busPtr, context, handler, connectLock, id);
+
+                AZ::Entity* entity = nullptr;
+                AZ::ComponentApplicationBus::BroadcastResult(entity, &AZ::ComponentApplicationBus::Events::FindEntity, id);
+                if (entity)
+                {
+                    // Only immediately dispatch if the entity is already active, otherwise when
+                    // entity will get activated it will make the notifications itself.
+                    const AZ::Entity::State entityState = entity->GetState();
+                    if (entityState == AZ::Entity::State::Active)
+                    {
+                        // Only immediately dispatch if the entity is a CharacterRequests' handler.
+                        CharacterRequestBus::EnumerateHandlersId(
+                            id,
+                            [&handler, id](const CharacterRequests* characterHandler)
+                            {
+                                if (characterHandler->IsPresent())
+                                {
+                                    handler->OnCharacterActivated(id);
+                                }
+                                else
+                                {
+                                    handler->OnCharacterDeactivated(id);
+                                }
+                                return true;
+                            });
+                    }
+                }
+            }
+        };
+
+    public:
+        /// With this connection policy, CharacterNotifications::OnCharacterActivated and
+        /// CharacterNotifications::OnCharacterDeactivated events will be immediately
+        /// dispatched when a handler connects to the bus.
+        template<class Bus>
+        using ConnectionPolicy = CharacterNotificationsConnectionPolicy<Bus>;
+
+        virtual ~CharacterNotifications() = default;
+
+        /// Notifies that the character controller has activated on an entity.
+        /// Other components that depend on the character controller can't simply rely on the component service
+        /// dependencies because the character controller is activated at OnEntityActivated, after the
+        /// components' activation.
+        virtual void OnCharacterActivated([[maybe_unused]] const AZ::EntityId& entityId)
```

Comment:
I'm wondering if there's any value in having the entity Id as a parameter here. Wouldn't you already need to know the entity Id in order to have connected to the bus?

Reply:
Good question. This is useful when it's required to know if more that 1 character is ready, so it can be easily used with a Multihandler and therefore it's very useful to know which one is calling.

Reply:
Yeah, it should be "gettable" inside the OnCharacterActivated() via "CharacterNotificationBus::GetCurrentBusId()". That's also admittedly a little bit less obvious and readable. I'm personally pretty ambivalent on this one, but that's why I didn't initially include the entityId in the first version of this event. :)

Reply:
Yeah, I don't think it's a big deal either way, I was just curious.